### ISSUE TITLE
Can't generate images base on images correct

### ISSUE THREAD

**Issue by illrayy** [UTokyo-FieldPhenomics-Lab/DODA#3]
Hello, as shown in the figure, in the image generated by generate_data_for_target_domain.py, the label position I want is very random and almost wrong. I want to know the reason
![dce3b4ab28a1d22d611633314237da2d](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/c4ed645c-2f38-495c-9c31-4579b8d57cf8)
![fc5746b9270ee68ed77d1b3334268fd1](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/28af6e2f-e736-44ef-b1e5-7d2a5d6fc437)

_Originally posted by @yangjie1874 in https://github.com/UTokyo-FieldPhenomics-Lab/DODA/issues/2#issuecomment-2209887718_

**Comment by illrayy** [comment]
@yangjie1874 It seems that your controlnet has not converged, or the Control Strength is set too small.
This is the image I got with the pre-trained DODA. For controlnet, how many iterations did you train?
![image](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/93829503/e7af01a7-0a79-4db2-b7ea-fe81ccf30e6b)

**Comment by yangjie1874** [comment]
sorry, I using ldm,  I have no training controlnet,

**Comment by yangjie1874** [comment]
just train ldm in the same way as #2

**Comment by illrayy** [comment]
DODA uses label (layout images in paper) to control the image generation, and the control achieved through controlnet(cldm), so if you only use ldm, it won't use layout images to control the generation, and the label won't match the image.

Training cldm requires boxes annotation, if you don't have the annotation, you can directly use the pre-trained weight to generate data. OR mix your data with GWHD to train the ldm first, then use the annotation from GWHD to train the cldm. As show in Table 3 and Figure 5 of the paper, although there is no annotation from the target domain, this training method can still improve the quality of the generated target domain data.

**Comment by yangjie1874** [comment]
Thank you very much for your patient answer. You mean I will now use the trained ldm as pre-trained weights to train cldm?

**Comment by illrayy** [comment]
for your dataset, do you have annotation?

**Comment by yangjie1874** [comment]
yes！

**Comment by yangjie1874** [comment]
The ldm I trained is unlabeled, but when I use the weights of ldm to train cldm, an error occurs. what should i do ? /(ㄒoㄒ)/~~
![0d2968cc40139d1149d64be2a3e714c6](https://github.com/UTokyo-FieldPhenomics-Lab/DODA/assets/143858218/c3ab3125-5ae9-40f1-8ccb-8775836f05d4)

**Comment by yangjie1874** [comment]
Or maybe I didn't notice some problems when I was training ldm?
